src/ml/neural_net/tf_activity_classifier_backend.cpp

```diff
@@ -0,0 +1,195 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+*/
+
+#include <ml/neural_net/tf_activity_classifier_backend.hpp>
+
+#include <iostream>
+#include <vector>
+
+#include <ml/neural_net/float_array.hpp>
+#include <ml/neural_net/tf_compute_context.hpp>
+#include <core/util/try_finally.hpp>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <pybind11/eval.h>
+
+namespace turi {
+namespace neural_net {
+
+namespace py = pybind11;
+
+using turi::neural_net::float_array;
+using turi::neural_net::float_array_map;
+using turi::neural_net::shared_float_array;
+
+
+template <typename CallFunc> 
+auto call_pybind_function(CallFunc&& func) -> decltype(func()) {
+    PyGILState_STATE gstate;
+  	gstate = PyGILState_Ensure();
+
+    turi::scoped_finally gstate_restore([&](){PyGILState_Release(gstate);});
+
+	try {
+		func();
+	} catch(...){
+	    	log_and_throw("An error occurred!");
+	    }
+}
+
+
+static std::vector<size_t> get_dimensions( const float_array& num) {
+  std::vector<size_t> result(num.dim());
+  std::memcpy(result.data(), num.shape(), num.dim()* sizeof(size_t));
+  return result;
+}
+
+static std::vector<size_t> get_strides( const float_array& num) {
+  std::vector<size_t> result(num.dim());
+  std::fill(result.begin(), result.end(), sizeof(float));
+  return result;
```

Comment:
I don't think this computation is correct, if the `float_array` has more than one dimension. For a `float_array` with shape `[3,3]`, for example, the strides are
`[12,4]`, since it takes 12 bytes to step from one row to the next (and 4 bytes to step from one column within one row to the next)

Reply:
Got it